supervisely/geometry/mask_3d.py

```diff
@@ -268,13 +268,19 @@ def from_file(figure, file_path: str):
         """
         mask3d_data, mask3d_header = nrrd.read(file_path)
         figure.geometry.data = mask3d_data
-        figure.geometry._space_origin = PointVolume(
-            x=mask3d_header["space origin"][0],
-            y=mask3d_header["space origin"][1],
-            z=mask3d_header["space origin"][2],
-        )
-        figure.geometry._space = mask3d_header["space"]
-        figure.geometry._space_directions = mask3d_header["space directions"]
+        try:
+            figure.geometry._space_origin = PointVolume(
+                x=mask3d_header["space origin"][0],
+                y=mask3d_header["space origin"][1],
+                z=mask3d_header["space origin"][2],
+            )
+            figure.geometry._space = mask3d_header["space"]
+            figure.geometry._space_directions = mask3d_header["space directions"]
+        except KeyError as e:
+            if str(e) in ("'space origin'", "'space'", "'space directions'"):
+                logger.debug(
+                    f"The Mask3D geometry for '{figure.parent_object.obj_class.name}' object doesn't contain optional space attributes"
```

Comment:
file_path

Reply:
will try to use space attributes later from the volume during upload?

Reply:
warn

Reply:
instead of file_path use figure ID